model/aqua-sim-mac-fama.cc

```diff
@@ -231,43 +260,89 @@ AquaSimFama::RecvProcess(Ptr<Packet> pkt)
       return false;
   }
 
-
+/*
   if( m_waitCTSTimer.IsRunning() ) {
-	  //printf("%f: node %d receive RTS\n", NOW, index_);
+      //printf("%f: node %d receive RTS\n", NOW, index_);
+      
+      //NS_LOG_INFO("time: " << Simulator::Now().GetSeconds() << "   node: " << AquaSimAddress::ConvertFrom(m_device->GetAddress()) << "receive RTS-------------------------------------");
+      NS_LOG_FUNCTION("CTSTimer.IsRunning()");
+      NS_LOG_FUNCTION("NextHop : " << asHeader.GetNextHop());
+      NS_LOG_FUNCTION("Address : " << m_device->GetAddress()); 
+     
+      NS_LOG_FUNCTION("ptag.GetPacketType(): " << ptag.GetPacketType());
+      NS_LOG_FUNCTION("FamaH.GetPType(): " << unsigned(FamaH.GetPType()));
+      NS_LOG_FUNCTION("FamaHeader::CTS" << FamaHeader::CTS);
+      NS_LOG_FUNCTION("m_device->GetAddress(): " << m_device->GetAddress());
+
       m_waitCTSTimer.Cancel();
+      
+
+      if(dst == m_device->GetAddress()) {
+        SendDataPkt();
+      }
+
       if(( ptag.GetPacketType() == AquaSimPtTag::PT_FAMA )&&(FamaH.GetPType()==FamaHeader::CTS)
-	      && (asHeader.GetNextHop() == m_device->GetAddress())) {
+	      && (dst == m_device->GetAddress())) {
 	  //receiving the CTS
+	  NS_LOG_INFO("Receiving the CTS");
+	 
 	  SendDataPkt();
       }
       else {
+          //NS_LOG_INFO("Before DoBackoff");
 	  DoBackoff();
       }
       pkt=0;
       return true;
   }
-
+*/
 
   if( ptag.GetPacketType() == AquaSimPtTag::PT_FAMA ) {
       switch( FamaH.GetPType() ) {
 	case FamaHeader::RTS:
 	  //printf("%f: node %d receive RTS\n", NOW, index_);
-	  if( dst == m_device->GetAddress() ) {
+	//  NS_LOG_INFO("time: " << Simulator::Now().GetSeconds() << "  node: " << AquaSimAddress::ConvertFrom(m_device->GetAddress()) << "receive RTS------------------------------------");
+
+
+          if( dst == m_device->GetAddress() ) {
 	      ProcessRTS(FamaH.GetSA());
 	  }
 	  DoRemote(m_CTSTxTime+2*m_maxPropDelay+m_estimateError);
 	  break;
 
 	case FamaHeader::CTS:
 	  //printf("%f: node %d receive CTS\n", NOW, index_);
-	  // this CTS must not be for this node
-	  DoRemote(2*m_maxPropDelay+m_estimateError);
-	  break;
+	  //NS_LOG_INFO("time: " << Simulator::Now().GetSeconds() << "  node: " << AquaSimAddress::ConvertFrom(m_device->GetAddress()) << "receive CTS------------------------------------");
+          if(m_waitCTSTimer.IsRunning()) {
```

Comment:
This section is actually meant to directly mirror past work of FAMA. While I don't recall the exact protocol or meanings behind it, under the FamaHeader::CTS case it is only meant to call the DoRemote function. Unless this is code breaking, this change should be reverted.

Reply:
Thanks for your reply.
in line 329, there is `doRemote`. i think if i erase line 328(`else`) it's no problem.
i removed this line because of duplicate of receiving CTS packet. so DoRemote has moved line 329 and i think else need to be removed.
please reply it. i'm really appreciate your help.

Reply:
Hanging code between the two break statements without a case condition is not advised. Please remove this entire part (lines 294-308).